components/eamxx/src/python/libpyscream/pyfield.hpp

```diff
@@ -24,7 +26,9 @@ struct PyField {
     f.allocate_view();
   }
 
-  pybind11::array get () const {
+  // FIXME: how to return ndarray?
+  template <typename X, typename Y, typename Z>
```

Comment:
This seems obscure. There's no way to understand what X/Y/Z mean here, and we only use one template overload anyways. Nanobind (like pybind) does not require to specify any template argument for ndarray. While template args may provide compile/runtime benefits (like type checking and certain compiler optimizations), given our use case, I would advocate for usable and generic code first (we can optimize perf later).

I think you can create an ndarray without template args, and pass data type, shape, and framework as constructor arguments.